File: src/main/java/com/io/klpn/match_statistic/enums/Event.java
package com.io.klpn.match_statistic.enums;

import java.util.Objects;
import java.util.stream.Stream;

public enum Event {

    GOAL("goal"),
    ASSIST("assist"),
    OWN_GOAL("own goal"),
    RED_CARD("red card"),
    YELLOW_CARD("yellow card");

    private String eventType;

    Event(String eventType) {
        this.eventType = eventType;
    }

    public String getEventType() {
        return this.eventType;
    }

    public static Event getEvent(String eventType) {
        return Stream.of(values())
                .filter(event -> event.eventType.equals(eventType))
                .findFirst()
                .orElseThrow(() -> new IllegalStateException(
                        String.format("eventType must be one of '%s', '%s', '%s', '%s' or '%s'",
                        GOAL.getEventType(), ASSIST.getEventType(), OWN_GOAL.getEventType(),
                                RED_CARD.getEventType(), YELLOW_CARD.getEventType())));
    }

    public boolean isGoal() {
        return Objects.equals(this.eventType, GOAL.getEventType());
    }

    public boolean isAssist() {
        return Objects.equals(this.eventType, ASSIST.getEventType());
    }

    public boolean isOwnGoal() {
        return Objects.equals(this.eventType, OWN_GOAL.getEventType());
    }

    public boolean isRedCard() {
        return Objects.equals(this.eventType, RED_CARD.getEventType());
    }

    public boolean isYellowCard() {
        return Objects.equals(this.eventType, YELLOW_CARD.getEventType());
    }

}

File: src/main/angular/src/app/match-edit/event.ts
export enum Event_m {
    GOAL = 'GOAL',
    ASSIST = 'ASSIST',
    OWN_GOAL = 'OWN_GOAL',
    RED_CARD = 'RED_CARD',
    YELLOW_CARD = 'YELLOW_CARD'
  }
  
  export function mapToEnumValue(str: string): Event_m {
  
    switch(str) {
      case 'Goal':
        return Event_m.GOAL;
      case 'Asysta':
        return Event_m.ASSIST;
      case 'Bramka samobójcza':
        return Event_m.OWN_GOAL;
      case 'Czerwona kartka':
        return Event_m.RED_CARD;
      case 'Żółta kartka':
        return Event_m.YELLOW_CARD;
      default:
          break;
    }
  
    return Event_m.GOAL;
  }